tests/unit/frameworks/dynamic_sparse/dynamic_network_construction_test.py

```diff
@@ -0,0 +1,102 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2019, Numenta, Inc.  Unless you have an agreement
+#  with Numenta, Inc., for a separate license for this software code, the
+#  following terms and conditions apply:
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see http://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+
+import unittest
+
+import torch
+from torch import nn
+
+from nupic.research.frameworks.dynamic_sparse.networks import (
+    DSConv2d,
+    DSLinear,
+    GSCSparseCNN,
+    init_coactivation_tracking,
+    make_dsnn,
+    squash_layers,
+    swap_layers,
+)
+from nupic.torch.modules import KWinners, KWinners2d, SparseWeights
+
+
+class NetworkConstructionTests(unittest.TestCase):
+    def setUp(self):
+        pass
+
+    def test_dynamic_gsc(self):
+
+        config = dict(
+            prune_methods=[None, "dynamic-conv", "dynamic-linear", None]
+        )
+
+        net = GSCSparseCNN()
+        make_dsnn(net, config)
+
+        dynamic_layers = []
+        for l in net.modules():
+            if isinstance(l, (DSLinear, DSConv2d)):
+                dynamic_layers.append(l)
+
+        assert len(dynamic_layers) == 2
```

Comment:
In unit tests, should use self.assertTrue or other unit test comparisons. See https://docs.python.org/3/library/unittest.html#assert-methods

Reply:
Fixed